src/main/java/org/jenkinsci/plugins/github/config/GitHubPluginConfig.java

```diff
@@ -111,60 +132,78 @@ public boolean isOverrideHookURL() {
-                HttpURLConnection con = (HttpURLConnection) new URL(value).openConnection();
-                con.setRequestMethod("POST");
-                con.setRequestProperty(GitHubWebHook.URL_VALIDATION_HEADER, "true");
-                con.connect();
-                if (con.getResponseCode() != 200) {
-                    return FormValidation.error("Got %d from %s", con.getResponseCode(), value);
-                }
-                String v = con.getHeaderField(GitHubWebHook.X_INSTANCE_IDENTITY);
-                if (v == null) {
-                    // people might be running clever apps that's not Jenkins, and that's OK
-                    return FormValidation.warning("It doesn't look like %s is talking to any Jenkins. " +
-                            "Are you running your own app?", value);
-                }
-                RSAPublicKey key = identity.getPublic();
-                String expected = new String(Base64.encodeBase64(key.getEncoded()));
-                if (!expected.equals(v)) {
-                    // if it responds but with a different ID, that's more likely wrong than correct
-                    return FormValidation.error("%s is connecting to different Jenkins instances", value);
-                }
-
-                return FormValidation.ok();
-            } catch (IOException e) {
-                return FormValidation.error(e, "Failed to test a connection to %s", value);
+        List<AbstractProject> registered = GitHubWebHook.get().reRegisterAllHooks();
+
+        LOGGER.info("Called registerHooks() for {} jobs", registered.size());
+        return FormValidation.ok("Called re-register hooks for %s jobs", registered.size());
+    }
+
+    @SuppressWarnings("unused")
+    public FormValidation doCheckHookUrl(@QueryParameter String value) {
+        try {
+            HttpURLConnection con = (HttpURLConnection) new URL(value).openConnection();
+            con.setRequestMethod("POST");
+            con.setRequestProperty(GitHubWebHook.URL_VALIDATION_HEADER, "true");
+            con.connect();
+            if (con.getResponseCode() != 200) {
+                return FormValidation.error("Got %d from %s", con.getResponseCode(), value);
             }
+            String v = con.getHeaderField(GitHubWebHook.X_INSTANCE_IDENTITY);
+            if (v == null) {
+                // people might be running clever apps that's not Jenkins, and that's OK
+                return FormValidation.warning("It doesn't look like %s is talking to any Jenkins. "
+                        + "Are you running your own app?", value);
+            }
+            RSAPublicKey key = identity.getPublic();
+            String expected = new String(Base64.encodeBase64(key.getEncoded()));
+            if (!expected.equals(v)) {
+                // if it responds but with a different ID, that's more likely wrong than correct
+                return FormValidation.error("%s is connecting to different Jenkins instances", value);
```

Comment:
Will fix it in separate pr. This code only reformatted after deletion of nested class